bouncer/tests/broker_fee_collection_test.ts

```diff
@@ -0,0 +1,251 @@
+  [Assets.BTC]: BigInt(190),
+  [Assets.USDC]: BigInt(0), // Fee is too low for localnet, it rounds to 0
+};
+const maxWithdrawalFee = {
+  [Assets.ETH]: BigInt(490000),
+  [Assets.DOT]: BigInt(197450000),
+  [Assets.FLIP]: BigInt(300000000),
+  [Assets.BTC]: BigInt(100),
+  [Assets.USDC]: BigInt(0),
+};
+const chainflip = await getChainflipApi();
+
+let withdrawLockout = false;
+let broker1: KeyringPair | undefined;
+export async function broker1KeyringPair() {
+  if (!broker1) {
+    await cryptoWaitReady();
+    const keyring = new Keyring({ type: 'sr25519' });
+    broker1 = keyring.createFromUri('//BROKER_1');
+  }
+  return broker1;
+}
+
+/// Runs a swap, checks that the broker fees are collected,
+/// then withdraws the broker fees, making sure the balance is correct after the withdrawal.
+async function testBrokerFees(asset: Asset, seed?: string): Promise<void> {
+  const broker = await broker1KeyringPair();
+
+  // Check the broker fees before the swap
+  const earnedBrokerFeesBefore = BigInt(
+    (await chainflip.query.swapping.earnedBrokerFees(broker.address, asset)).toString(),
+  );
+  console.log(`${asset} earnedBrokerFeesBefore:`, earnedBrokerFeesBefore);
+
+  // Run a swap
+  const swapAsset = asset === Assets.USDC ? Assets.FLIP : Assets.USDC;
+  const destinationAddress = await newAddress(swapAsset, seed ?? randomBytes(32).toString('hex'));
+  const useableDestinationAddress =
+    asset === Assets.DOT ? decodeDotAddressForContract(destinationAddress) : destinationAddress;
+  const destinationChain = chainShortNameFromAsset(swapAsset); // "ETH" -> "Eth"
+  console.log(`${asset} destinationAddress:`, destinationAddress);
+  const observeSwapScheduledEvent = observeEvent(
+    ':SwapScheduled',
+    chainflip,
+    (event) =>
+      event.data.destinationAddress[destinationChain]?.toLowerCase() ===
+      useableDestinationAddress.toLowerCase(),
+  );
+  console.log(`Running swap ${asset} -> ${swapAsset}`);
+  await performSwap(
+    asset,
+    swapAsset,
+    destinationAddress,
+    undefined,
+    undefined,
+    undefined,
+    testSwapAssetAmount[asset].toString(),
+    testCommissionBps,
+    false,
+  );
+
+  // Check the broker fees after the swap
+  const earnedBrokerFeesAfter = BigInt(
+    (await chainflip.query.swapping.earnedBrokerFees(broker.address, asset)).toString(),
+  );
+  console.log(`${asset} earnedBrokerFeesAfter:`, earnedBrokerFeesAfter);
+
+  // Get values from the swap event
+  const swapScheduledEvent = await observeSwapScheduledEvent;
+  const brokerCommission = BigInt(
+    JSON.stringify(swapScheduledEvent.data.brokerCommission)
+      .replaceAll('"', '')
+      .replaceAll(',', ''),
+  );
+  console.log('brokerCommission:', brokerCommission);
+
+  // Check that the deposit amount is correct after deducting the deposit fee
+  const depositAmount = BigInt(
+    JSON.stringify(swapScheduledEvent.data.depositAmount).replaceAll('"', '').replaceAll(',', ''),
+  );
+  const testSwapAmount = BigInt(
```

Comment:
A nit, but why "test" in the name? Everything here is for a test, so I'm not sure if this adds anything to clarity.